Update dependency bounds for llmcompressor 0.9 release

dhuangnm opened the pull request:
SUMMARY:
Update dependency bounds based on their latest releases.


TEST PLAN:
All nightly tests.

Run with upper bounds: https://github.com/neuralmagic/llm-compressor-testing/actions/runs/19864870462
Run with lower bounds: https://github.com/neuralmagic/llm-compressor-testing/actions/runs/19867046299

dhuangnm commented on setup.py:
```diff
@@ -138,13 +139,13 @@ def localversion_func(version: ScmVersion) -> str:
             else "compressed-tensors>=0.12.3a2"
         ),
-        ("auto-round==0.9.1"),
+        ("auto-round==0.9.1" if BUILD_TYPE == "release" else "auto-round==0.9.1"),
```
Updated, I'll file another PR for the workflows to use the new [autoround] install tag.

dhuangnm replied:
Updated, and allowed latest `auto-round` 0.9.2, also reverted changes to `accelerator`.

dsikka commented:
> > Thanks!
> > Just an FYI - I think we want to pin the transformers version, regardless of release. This is for the time being while we ensure we can support the v5 release that will be out soon.
> 
> Got it, updated the upper bound for transformers to 5.0.0 although currently its latest is 5.0.0.rc0 on the PyPi.

Oh sorry, I meant the upper bound should be the current stable release

dsikka commented on setup.py:
```diff
@@ -138,15 +139,17 @@ def localversion_func(version: ScmVersion) -> str:
-        ("auto-round==0.9.1"),
     ],
     extras_require={
+        "autoround": [
```
Thanks! We will also need to change the transformers test workflow to run autoround tests as they require this install:

https://github.com/vllm-project/llm-compressor/blob/6284e1bd7d690fc0e3eb713194b061bc14517de4/.github/workflows/test-check-transformers.yaml#L104

If we think this will be a bigger change, we can make the autoround optional install change in a followup

dsikka replied:
Should be a separate job, no?
Sorry, I think my request is becoming larger than we expected. We can revert to install autoround how you were previously (i.e not using the optional install) and I'll add a follow-up to fix this

dhuangnm commented:
> > > Thanks!
> > > Just an FYI - I think we want to pin the transformers version, regardless of release. This is for the time being while we ensure we can support the v5 release that will be out soon.
> > 
> > 
> > Got it, updated the upper bound for transformers to 5.0.0 although currently its latest is 5.0.0.rc0 on the PyPi.
> 
> Oh sorry, I meant the upper bound should be the current stable release

Gotcha, updated.

hshen14 commented:
Hi @dsikka @dhuangnm  anything that we need to follow up with AutoRound, e.g., transformer version? We can try aligning with LLM-C's dependencies at AutoRound side to make the integration easier.

dsikka commented on setup.py:
```diff
@@ -114,22 +114,24 @@ def localversion_func(version: ScmVersion) -> str:
+        ("torch>=2.7.0,<=2.9.1" if BUILD_TYPE == "release" else "torch>=2.7.0"),
         (
-            "transformers>=4.54.0,<=4.56.2"
+            "transformers>=4.54.0,<=4.57.3"
```
We want to cap transformers to the latest stable release for both BUILD_TYPE == "release" and  BUILD_TYPE != "release"

dhuangnm replied:
Updated and limit to "transformers>=4.54.0,<=4.56.2" for both release and nightly for now.

kylesayrs reviewed (approved):
Thanks

dsikka commented on setup.py:
```diff
@@ -114,22 +114,27 @@ def localversion_func(version: ScmVersion) -> str:
-            "accelerate>=1.6.0,<=1.10.1"
+            "auto-round>=0.9.1,<=0.9.2"
+            if BUILD_TYPE == "release"
+            else "auto-round>=0.9.1"
```
Will the autoround lower bound work with our accelerate dependency?

dhuangnm replied:
Oh good catch, yes the 0.9.1 will only work with `accelerate` 1.10.0. Should we only allow `auto-round >=0.9.2`?

dhuangnm replied:
Thanks, updated.
